src/metabase/pulse/render/body.clj

```diff
@@ -247,15 +247,19 @@
                :quarter "YYYY - [Q]Q"}
    "MMMM D, YYYY" {:month "MMMM, YYYY"}
    "D MMMM, YYYY" {:month "MMMM, YYYY"}
-   "dddd, MMMM D, YYYY" {:week "MMMM D, YYYY"
+   "dddd, MMMM D, YYYY" {:day "EEEE, MMMM d, YYYY"
+                         :week "MMMM d, YYYY"
                          :month "MMMM, YYYY"}})
 
 (defn- update-date-style
-  [date-style unit]
+  [date-style unit {::mb.viz/keys [date-abbreviate date-separator]}]
   (let [unit (or unit :default)]
-    (or (get-in override-date-styles [date-style unit])
-        (get-in default-date-styles [unit])
-        date-style)))
+    (cond-> (or (get-in override-date-styles [date-style unit])
+                (get-in default-date-styles [unit])
+                date-style)
+      (not= date-separator "/") (str/replace #"/" date-separator)
+      date-abbreviate (-> (str/replace #"MMMM" "MMM")
+                          (str/replace #"EEEE" "E")))))
```

Comment:
If the static-viz settings have `{:date-abbreviate true}`, we want to adjust the formatter to respect this. String replacing these covers such a case.